src/materials/ShaderMaterial.js

```diff
@@ -116,7 +116,31 @@ ShaderMaterial.prototype.toJSON = function ( meta ) {
 
 	var data = Material.prototype.toJSON.call( this, meta );
 
-	data.uniforms = this.uniforms;
+	data.uniforms = {};
+
+	for ( var name in this.uniforms ) {
+
+		var uniform = this.uniforms[ name ];
+
+		if ( uniform.value.isTexture ) {
+
+			var texture = uniform.value;
+
+			data.uniforms[ name ] = {
+				type: 't',
+				value: texture.toJSON( meta ).uuid
+			};
+
+		} else {
+
+			data.uniforms[ name ] = {
+				value: uniform.value
```

Comment:
Don't we need `value` serialization here? I think `value` can be Three.js object, like `THREE.Color`, `THREE.Vector3`, `THREE.Matrix`, and so on.

Reply:
Okay, let's expand the PR a bit. I will add support for common uniform types.